actionpack/lib/action_dispatch/middleware/actionable_exceptions.rb

```diff
@@ -19,6 +19,9 @@ def call(env)
       ActiveSupport::ActionableError.dispatch(request.params[:error].to_s.safe_constantize, request.params[:action])
 
       redirect_to request.params[:location]
+    rescue Exception => error
+      ActiveSupport::ActionableError.trigger_by(error)
```

Comment:
The middleware tries to check whether the raised error can trigger an actionable error. Only one actionable error can be triggered at a time, as this just reraises an actionable error from the current one.

The nice part about 3rd party plugins defining actionable errors is that they don't need to care how to raise it if they define a trigger. The actionable exceptions middleware is in the stack anyway at this point.